server/svix-server/src/test_util.rs

```diff
@@ -0,0 +1,232 @@
+            .context("error polling endpoint")?;
+
+        if status != expected_code {
+            anyhow::bail!(
+                "assertation failed: expected {}, received {}",
+                expected_code,
+                status
+            )
+        }
+
+        Ok(())
+    }
+
+    pub async fn asserting_get<O: DeserializeOwned + Clone + PartialEq + std::fmt::Debug>(
+        &self,
+        endpoint: &str,
+        expected_code: StatusCode,
+        expected_resp: Option<O>,
+    ) -> Result<()> {
+        if let Some(expected_resp) = expected_resp {
+            self.asserting_request::<(), O, _>(
+                endpoint,
+                HashMap::new(),
+                Method::Get,
+                None,
+                expected_code,
+                move |actual_resp| {
+                    if actual_resp == expected_resp {
+                        Ok(())
+                    } else {
+                        anyhow::bail!("exppected {:?}, received {:?}", expected_resp, actual_resp)
+                    }
+                },
+            )
+            .await
+            .map(|_| ())
+        } else {
+            self.asserting_request_no_response_body::<()>(
+                endpoint,
+                HashMap::new(),
+                Method::Get,
+                None,
+                expected_code,
+            )
+            .await
+        }
+    }
+
+    pub async fn asserting_post<
+        I: Serialize,
+        O: DeserializeOwned + Clone,
+        F: FnOnce(O) -> Result<()>,
+    >(
+        &self,
+        endpoint: &str,
+        input: I,
+        expected_code: StatusCode,
+        response_assertation: F,
+    ) -> Result<O> {
+        self.asserting_request(
+            endpoint,
+            HashMap::new(),
+            Method::Post,
+            Some(input),
+            expected_code,
+            response_assertation,
+        )
+        .await
+    }
+
+    pub async fn asserting_put<
+        I: Serialize,
+        O: DeserializeOwned + Clone,
+        F: FnOnce(O) -> Result<()>,
+    >(
+        &self,
+        endpoint: &str,
+        input: I,
+        expected_code: StatusCode,
+        response_assertation: F,
+    ) -> Result<O> {
+        self.asserting_request(
+            endpoint,
+            HashMap::new(),
+            Method::Put,
+            Some(input),
+            expected_code,
+            response_assertation,
+        )
+        .await
+    }
+
+    pub async fn asserting_delete(&self, endpoint: &str, expected_code: StatusCode) -> Result<()> {
+        self.asserting_request_no_response_body::<()>(
+            endpoint,
+            HashMap::new(),
+            Method::Delete,
+            None,
+            expected_code,
+        )
+        .await
+    }
+}
+
+pub fn start_svix_server() -> (TestClient, tokio::task::JoinHandle<()>) {
+    let _ = dotenv::dotenv();
+    let cfg = crate::cfg::load().unwrap();
+    let token = crate::generate_token(&cfg.jwt_secret).unwrap();
```

Comment:
You probably want to generate a different token for each test each with a random OrganizationId. This way tests will be completely isolated from one another rather than reusing the same account (so we can potentially run them in parallel!).